Describe Sonos media_player.play_media support (#18202)
home-assistant/home-assistant.github.io: source/_integrations/sonos.markdown

The Sonos integration adds one `switch` for each alarm set in the Sonos app. The alarm switches are detected, deleted and assigned automatically and come with several attributes that help to monitor Sonos alarms.

## Playing media

> Sonos accepts a variety of `media_content_id` formats in the `media_player.play_media` service, but most commonly as URIs. For example, both Spotify and Tidal share links can be provided as-is. Playback of [music hosted on a Plex server](https://www.home-assistant.io/integrations/plex#sonos-playback) is possible. Direct HTTP/HTTPS links to local or remote media files can also be used if the Sonos device can reach the URI directly, but specific media encoding support may vary.

Reviewer: ```suggestion
Sonos accepts a variety of `media_content_id` formats in the `media_player.play_media` service, but most commonly as URIs. For example, both Spotify and Tidal share links can be provided as-is. Playback of [music hosted on a Plex server](/integrations/plex#sonos-playback) is possible. Direct HTTP/HTTPS links to local or remote media files can also be used if the Sonos device can reach the URI directly, but specific media encoding support may vary.
```

Spotify share links can also be provided in the `spotify:playlist:abcdefghij0123456789XY` format.

> An optional `enqueue` argument can be added to the service call. If `True`, the media will be appended to the end of the playback queue. If not provided or `False` then the queue will be replaced

Reviewer: ```suggestion
An optional `enqueue` argument can be added to the service call. If `true`, the media will be appended to the end of the playback queue. If not provided or `false` then the queue will be replaced.
```

> ### Examples:

Reviewer: ```suggestion
### Examples:

This is an example service call that plays an audio file, from a web server on the local network (like the Home Assistant built-in webserver):

```

Author: ```suggestion
### Examples:

This is an example service call that plays an audio file from a web server on the local network (like the Home Assistant built-in webserver):

```

```yaml

> # Play an audio file from the local network:

Reviewer: ```suggestion
```

service: media_player.play_media
target:
  entity_id: media_player.sonos
data:
  media_content_type: "music"
  media_content_id: "http://192.168.1.50:8123/local/sound_files/doorbell-front.mp3"
```

> 

Reviewer: ```suggestion

Sonos can also play music from Spotify. The Sonos integration accepts and can play Spotify URLs directly. An example service call:

```

Author: ```suggestion

Sonos can also play music or playlists from Spotify if linked in Sonos. Full Spotify URIs and Spotify ID codes can both be used directly. An example service call:

```

```yaml

> # Play a Spotify playlist

Reviewer: ```suggestion
```

service: media_player.play_media
target:
  entity_id: media_player.sonos
data:
  media_content_type: "playlist"
  media_content_id: "spotify:playlist:abcdefghij0123456789XY"
  enqueue: true
```

> 

Reviewer: Run a [Plex Media Server](/integrations/plex) in your home? The Sonos integration can work with that as well. This example play music originating from your Plex system:

Author: ```suggestion

Run a [Plex Media Server](/integrations/plex#sonos-playback) in your home? The Sonos integration can work with that as well. This example plays music directly from your Plex server:

```

```yaml

> # Play a Plex album (see link above for requirements)

Reviewer: ```suggestion
```

## Playing media

Sonos accepts a variety of `media_content_id` formats in the `media_player.play_media` service, but most commonly as URIs. For example, both Spotify and Tidal share links can be provided as-is. Playback of [music hosted on a Plex server](/integrations/plex#sonos-playback) is possible. Direct HTTP/HTTPS links to local or remote media files can also be used if the Sonos device can reach the URI directly, but specific media encoding support may vary.

Music services which require an account (e.g., Spotify) must first be configured using the Sonos app.

An optional `enqueue` argument can be added to the service call. If `true`, the media will be appended to the end of the playback queue. If not provided or `false` then the queue will be replaced.

### Examples:

This is an example service call that plays an audio file from a web server on the local network (like the Home Assistant built-in webserver):

```yaml
service: media_player.play_media
target:
  entity_id: media_player.sonos
data:
  media_content_type: "music"
  media_content_id: "http://192.168.1.50:8123/local/sound_files/doorbell-front.mp3"
```

> Sonos can also play music or playlists from Spotify. Full Spotify URIs and ID codes can both be used directly. An example service call:

Reviewer: This is incorrect now. Spotify ID codes are not usable, the example shows a Spotify URI. For more information, see the Spotify developers documentaion.

Author: Wrong terminology. Looks like I mean URIs and URLs, but not sure if that will be obvious to users. Maybe a second Spotify example would help.

> Sonos can also play music or playlists from Spotify. Full Spotify URIs and ID codes can both be used directly. An example service call:

Author: ```suggestion
Sonos can also play music or playlists from Spotify. Both Spotify URIs and URLs can be used directly. An example service call using a playlist URI:
```

```yaml
service: media_player.play_media
target:
  entity_id: media_player.sonos
data:
  media_content_type: "playlist"
  media_content_id: "spotify:playlist:abcdefghij0123456789XY"
  enqueue: true
```

> 

Author: ````suggestion

An example service call using a Spotify URL:

```yaml
service: media_player.play_media
target:
  entity_id: media_player.sonos
data:
  media_content_type: "music"
  media_content_id: "https://open.spotify.com/album/abcdefghij0123456789YZ"
```

````

Spotify share links can also be provided in the `spotify:playlist:abcdefghij0123456789XY` format.

| Service data attribute | Optional | Description                                                                                                                                                            |
| -----------------------| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `entity_id`            |       no | The `entity_id` of the target Sonos device. |
| `media_content_id`     |       no | A media identifier, normally in a URI format. Must be reachable directly from the Sonos device. |
| `media_content_type`   |       no | Should be one of `music`, `track`, or `playlist`. |

> | `enqueue`              |      yes | Set to `True` to add the media to the end of the playback queue. If empty or `False` the queue will be replaced. |

Reviewer: This is not part of this integration, but part of `media_player`.

Author: But the payloads for this service are unfortunately integration-specific.

Reviewer: I understand, we should/could add examples, but we should not add service descriptions from other integrations. Especially, since we plan on replacing those with the ones from the core code base (so they don't have to be maintained in here).
